components/graph/git-node.js

```diff
@@ -141,22 +145,28 @@ class GitNodeViewModel extends Animateable {
     } else {
       this.r(15);
       this.cx(610 + 90 * this.branchOrder());
-      this.cy(this.aboveNode ? this.aboveNode.cy() + 60 : 120);
+      this.cy(this.aboveNode && !isNaN(this.aboveNode.cy()) ? this.aboveNode.cy() + 60 : 120);
     }
 
     if (this.aboveNode && this.aboveNode.selected()) {
       this.cy(this.aboveNode.cy() + this.aboveNode.commitComponent.element().offsetHeight + 30);
     }
 
     this.color(this.ideologicalBranch() ? this.ideologicalBranch().color : '#666');
+    if (!this.hasBeenRenderedBefore) {
+      // push this nodes into the graph's node list to be rendered if first time.
+      // if been pushed before, no need to add to nodes.
+      this.hasBeenRenderedBefore = true;
+      this.graph.nodes.push(this);
```

Comment:
I don't understand how this can push here when it wasn't rendered before and there's no removal of a push elsewhere.

When I amend the latest commit, the current branch no longer shows up and I need to reload the page. Is that due to this?